File: GrooveOn.Models/ResponseObjects/AlbumResponse.cs
using System;

namespace GrooveOn.Model.Responses
{
    public class AlbumResponse
    {
        public int Id { get; set; }

        public string? ExternalAlbumId { get; set; }

        public string Source { get; set; } = string.Empty;

        public string Title { get; set; } = string.Empty;

        public int ArtistId { get; set; }

        public string ArtistName { get; set; } = string.Empty;

        public DateTime? ReleaseDate { get; set; }

        public string? CoverUrl { get; set; }

        public string? Description { get; set; }

        public DateTime CreatedAt { get; set; }

        public int SongCount { get; set; }
    }
}

File: UI/grooveon_mobile/lib/models/album_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'album_response.g.dart';

@JsonSerializable()
class AlbumResponse {
  final int id;
  final String? externalAlbumId;
  final String source;
  final String title;
  final int artistId;
  final String artistName;
  final DateTime? releaseDate;
  final String? coverUrl;
  final String? description;
  final DateTime createdAt;
  final int songCount;

  AlbumResponse({
    required this.id,
    this.externalAlbumId,
    required this.source,
    required this.title,
    required this.artistId,
    required this.artistName,
    this.releaseDate,
    this.coverUrl,
    this.description,
    required this.createdAt,
    required this.songCount,
  });

  factory AlbumResponse.fromJson(Map<String, dynamic> json) =>
      _$AlbumResponseFromJson(json);

  Map<String, dynamic> toJson() => _$AlbumResponseToJson(this);
}